blocks/media-upload-button/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { __ } from 'i18n';
+import Button from 'components/button';
+
+class MediaUploadButton extends Component {
+	constructor( { multiple = false, type } ) {
+		super( ...arguments );
+		this.openModal = this.openModal.bind( this );
+		this.onSelect = this.onSelect.bind( this );
+		const frameConfig = {
+			title: __( 'Select or Upload a media' ),
+			button: {
+				text: __( 'Select' ),
+			},
+			multiple,
+		};
+		if ( !! type ) {
+			frameConfig.library = { type };
+		}
+		this.frame = wp.media( frameConfig );
```

Comment:
Do we need to clean up this frame when the component unmounts? Maybe via the `view.remove()` function?

https://core.trac.wordpress.org/browser/trunk/src/wp-includes/js/media/views/view.js?rev=33337#L52